### PR TITLE
Refactor Managed Identity  unit tests

### PR DESCRIPTION
**Pull request by Avery-Dunn**
The ManagedIdentityTests class was made to test all managed identity behavior in one place, but as new sources were supported and new edge cases were tested the class ballooned in size.

This PR is a complete refactor of that test class, and does the following:
- Move tests into nested classes, to better organize similar test cases and allow them to be easily split off if they grow too large
- Create helper methods for common configurations and behaviors
- Move constants to their own file and follow standard naming

This refactor also tries to use the best practices and features of JUnit which were not available in the version the library was originally written. Ideally this will be an example that other test classes can follow to be more readable, maintainable, and expandable

### CODE REVIEW

File: msal4j-sdk/src/test/java/com/microsoft/aad/msal4j/ManagedIdentityTestConstants.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    // Endpoints
+    static final String APP_SERVICE_ENDPOINT = "http://127.0.0.1:41564/msi/token";
+    static final String IMDS_ENDPOINT = "http://169.254.169.254/metadata/identity/oauth2/token";
+    static final String AZURE_ARC_ENDPOINT = "http://localhost:40342/metadata/identity/oauth2/token";
+    static final String CLOUDSHELL_ENDPOINT = "http://localhost:40342/metadata/identity/oauth2/token";
+    static final String SERVICE_FABRIC_ENDPOINT = "http://localhost:40342/metadata/identity/oauth2/token";
+
+    // Example responses
+    static final String SUCCESSFUL_RESPONSE_INVALID_JSON = "missing starting bracket \"access_token\":\"accesstoken\",\"token_type\":" + "\"Bearer\",\"client_id\":\"a bunch of problems}";
```
Comment (gladjohn) [inline msal4j-sdk/src/test/java/com/microsoft/aad/msal4j/ManagedIdentityTestConstants.java]:
```suggestion
    static final String RESPONSE_MALFORMED_JSON = "missing starting bracket \"access_token\":\"accesstoken\",\"token_type\":" + "\"Bearer\",\"client_id\":\"a bunch of problems}";
```

Comment (gladjohn) [reply msal4j-sdk/src/test/java/com/microsoft/aad/msal4j/ManagedIdentityTestConstants.java]: just that the name is contradicting.

File: msal4j-sdk/src/test/java/com/microsoft/aad/msal4j/ManagedIdentityTestConstants.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.aad.msal4j;
+
+class ManagedIdentityTestConstants {
```
Comment (gladjohn) [inline msal4j-sdk/src/test/java/com/microsoft/aad/msal4j/ManagedIdentityTestConstants.java:6]: isn't there any utils where this can go?

**Review by gladjohn — APPROVED** [review]

**Review by bgavrilMS — APPROVED** [review]

File: msal4j-sdk/src/test/java/com/microsoft/aad/msal4j/ManagedIdentityTestConstants.java (hunk 2 above)
Comment (Avery-Dunn) [reply msal4j-sdk/src/test/java/com/microsoft/aad/msal4j/ManagedIdentityTestConstants.java:6]: There's a `Constants` file in the main library and one in the integration tests folder, but there wasn't one for the unit tests. The unit tests have `TestHelper`, but that's more for helper methods than constants.

I thought about making a more generic constants file and that might happen in a later refactor, but since these are only used for MI tests I figured it'd be easier to keep it more focused for this PR.

**Review by gladjohn — APPROVED** [review]
